### ISSUE TITLE
System command to find the issues on the filesystem layout

### ISSUE THREAD

**Issue by filimonov** [ClickHouse/ClickHouse#61229]
Detect wrong permissions / owners / presence of data files without metadata files, incorrect / missing symlinks.

**Comment by filimonov** [comment]
some strange examples: 

1. 

> During our investigation, we discovered that ClickHouse could not access certain directories within /var/lib/clickhouse/store due to the directories having only r------- permissions, thereby preventing ClickHouse from reading the folder contents.
>
> it seems to start happening after clickhouse crash (looks like caused by a memory damage), and some interactions with filesystem which were happening diring the crash
>  <Error> tenant_tmp.activity_bundle_RsH6i_202402 (0624ad4c-cf0b-4a54-9b45-4826f747778e): Cannot quickly remove directory /var/lib/clickhouse/^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@ by removing files; fallback to recursive removal. Reason: Code: 458. DB::ErrnoException: Cannot unlink file /var/lib/clickhouse/^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@^@, errno: 21, strerror: Is a directory. (CANNOT_UNLINK)

2. DBA  by accident did `chown 755` instead of `chmod 755` during some maintanance operations, so the file owner was user with id 755 instead of clickhouse.

3. relatevely often you can't remove the database, becase there are some leftovers in data folder from tables which does not exists for a long time already (don't know the exact scenario of how it happen).

**Comment by geldot** [comment]
Yes. It would be great to have a proper implementation of the logic in the official ClickHouse Docker image's [entrypoint.sh](https://github.com/ClickHouse/ClickHouse/blob/ad363fe2541096bf27cce0681ed6e62b2e1a580c/docker/server/entrypoint.sh). That file runs after each restart on prod and gives me nightmares. We're never sure what it's doing, and it makes changes silently without prompting or logging.

It looks like an (unsafe) attempt to address the issue of incorrect filesystem permissions. A better option would be to just bail out if there is an issue and allow the administrator to deal with the situation, as it could indicate a deeper issue such as an incorrect or failed mount or corruption.
